### PR TITLE
Mw/mirage/mw 353 entity

### PR DESCRIPTION
**Pull request by SANTOSH-MWP**
please add folderID in app-config.yaml.

**bigqueryapi:
  projectId: projectxxxxxx
  gbdataSet: xxxxx
  gbfolderId: "folders/74913165xxx"**

### CODE REVIEW

File: plugins/bigquery-api-backend/src/service/router.ts [hunk 1]
```diff
@@ -38,6 +38,7 @@ export async function createRouter(
   const projectId = readAppConfig.getString('projectId');
   const gbdataset = readAppConfig.getString('gbdataSet');
   const { GoogleAuth, Impersonated } = require('google-auth-library');
+  const gbfolderId = readAppConfig.getString('gbfolderId');
```
Comment (fosterm-mw) [inline plugins/bigquery-api-backend/src/service/router.ts:41]: We want to be able to get the folderId from the entity in Backstage. This should not come from the app-config but is fine for right now.

File: plugins/costinsight/src/components/CostapiComponent/CostapiComponent.js [hunk 2]
```diff
@@ -0,0 +1,382 @@
...
+        const getData = async () => {
+
+            const url = `${apiUrl}/api/bigqueryapi/dataset`
+            try {
+                const response = await fetch(url, {
+                    method: 'GET',
+                    headers: {
+                        'Content-type': 'application/json; charset=UTF-8',
+                    },
+                });
```
Comment (fosterm-mw) [inline plugins/costinsight/src/components/CostapiComponent/CostapiComponent.js:28]: We will need to add the folder ID to the GET header or body to query the correct folder id based on the specific entity.

**Review by fosterm-mw — APPROVED** [review]
Noting these changes, I can approve. I think the enhancement of ingesting the entity name would be important.

File: plugins/bigquery-api-backend/src/service/router.ts (hunk 1 above)
Comment (SANTOSH-MWP) [reply plugins/bigquery-api-backend/src/service/router.ts:41]: Okay, its hardly matter from where we are supplying params.
we can do it later.

File: plugins/costinsight/src/components/CostapiComponent/CostapiComponent.js (hunk 2 above)
Comment (SANTOSH-MWP) [reply plugins/costinsight/src/components/CostapiComponent/CostapiComponent.js:28]: will connect sometime to understand and make the changes appropriately.

**Pull request merged**